### ISSUE TITLE
3d gui

### ISSUE THREAD

**Issue by DuFF14** [OSVR/OSVR-Unity#40]
It would be great to have a 3d gui menu component similar to OVRMainMenu in the Oculus SDK for displaying FPS and other stats/configuration information.

**Comment by peroht** [comment]
I've done a rough csharp UI Canvas implementation for 5.3.1f1. The same gameobject hierachy does not seem to work in 5.2.1f1. 
It renders to VRSurface0 (meaning left eye only). How can I make it render to the right eye as well?

Is this the correct repo to fork and add to?

**Comment by DuFF14** [comment]
Yes, this is the correct repo. If it's only visible in one eye, does that mean the UI Canvas is in camera space and not world space? If it's in world space, which I think is a best practice for VR, it should be visible to both eyes. Support for multiple layers, such as a UI layer, is something we've talked about adding to RenderManager. If you do need a screen-space / UI overlay layer, implementing (or waiting for) that feature in RenderManager might be the best solution.

**Comment by peroht** [comment]
Yes i did it in camera  space. I'll change that and see if I can expand it a little. Is it OK if I use a prefab? (It's not a self contained source that builds up the whole UI by itself now).Or should I try and do it through scripting to keep the amount of prefabs low?


-------- Originalmeddelande --------
Från: Greg Aring <notifications@github.com> 
Datum: 2016-02-10  21:31  (GMT+01:00) 
Till: OSVR/OSVR-Unity <OSVR-Unity@noreply.github.com> 
Kopia: Peter Thor <peter.thor@vicator.com> 
Rubrik: Re: [OSVR-Unity] 3d gui (#40) 

Yes, this is the correct repo. If it's only visible in one eye, does that mean the UI Canvas is in camera space and not world space? If it's in world space, which I think is a best practice for VR, it should be visible to both eyes. Support for multiple layers, such as a UI layer, is something we've talked about adding to RenderManager. If you do need a screen-space / UI overlay layer, implementing (or waiting for) that feature in RenderManager might be the best solution.

**Comment by DuFF14** [comment]
I think a prefab would be good. Looking forward to seeing it.
